### PR TITLE
fix: selecting all pages with advanced/detail search corrected

### PR DESCRIPTION
**Pull request by headri**
this.state.currentSearchSelection is not set in detail / advanced search
therefore condition removed; the other condition is sufficient
closes https://github.com/ComPlat/chemotion/issues/217

### CODE REVIEW

File: app/packs/src/stores/alt/stores/UIStore.js [hunk 1]
```diff
@@ -188,31 +188,29 @@ class UIStore {
   handleCheckAllElements(params) {
     this.waitFor(ElementStore.dispatchToken);
 
-    let { type, range } = params;
-    let { elements } = ElementStore.getState();
+    const { type, range } = params;
+    const { elements } = ElementStore.getState();
 
-    if (range == 'all') {
-      if (this.state.currentSearchSelection && elements[type + "s"].ids) {
```
Comment (headri) [inline app/packs/src/stores/alt/stores/UIStore.js:195]: relevant change

Comment (JanCBrammer) [reply app/packs/src/stores/alt/stores/UIStore.js:195]: Just to be sure, the only behavioral change in this file's diff is the removal of `this.state.currentSearchSelection` from the guard clause? For the future please change behavior and formatting in separate commits.

**Review by JanCBrammer — CHANGES_REQUESTED** [review]
If this is a bug fix, please add a test that replicates (or at least approximates) the bug and demonstrates how this PR fixes it.

**Comment by PiTrem** [comment]
to be ported to main with test

**Pull request merged**